searchlib/src/tests/queryeval/iterator_benchmark/benchmark_blueprint_factory.cpp

```diff
@@ -159,24 +159,30 @@ class MyFactory : public BenchmarkBlueprintFactory {
 public:
     MyFactory(const FieldConfig& field_cfg, QueryOperator query_op,
               uint32_t num_docs, uint32_t default_values_per_document,
-              double op_hit_ratio, uint32_t children);
+              double op_hit_ratio, uint32_t children, bool disjunct_children);
 
     std::unique_ptr<Blueprint> make_blueprint() override;
 };
 
 MyFactory::MyFactory(const FieldConfig& field_cfg, QueryOperator query_op,
                      uint32_t num_docs, uint32_t default_values_per_document,
-                     double op_hit_ratio, uint32_t children)
+                     double op_hit_ratio, uint32_t children, bool disjunct_children)
     : _query_op(query_op),
       _docid_limit(num_docs + 1),
       _terms(),
       _searchable()
 {
     uint32_t hits_per_term = calc_hits_per_term(num_docs, op_hit_ratio, children, query_op);
     HitSpecs hit_specs(55555);
-    hit_specs.add(default_values_per_document, num_docs);
+    if (!disjunct_children) {
+        hit_specs.add(default_values_per_document, num_docs);
+    }
     _terms = hit_specs.add(children, hits_per_term);
-    _searchable = make_searchable(field_cfg, num_docs, hit_specs);
+    if (disjunct_children) {
+        uint32_t op_num_hits = num_docs * op_hit_ratio;
+        hit_specs.add(default_values_per_document, num_docs - op_num_hits);
```

Comment:
why less here?

Reply:
This ensures that the remaining docids are populate with a "default value". Only a single default value is supported when using disjunct children, so I have altered the code to reflect that.